src/test/java/com/joyent/manta/archiver/ManataClientSupplierTest.java

```diff
@@ -0,0 +1,296 @@
+                    + "aVf4Q0oP46MBiRIQgWvMraYJtMafssnSpz/Dt3AZAoGBAJ6vGVpqPbQ2Djohwzfp\n"
+                    + "vtPiYO6ezFC3S42iyzTEqy+yMJV+KwE7oKxlQdoGyqCM80TMxcjeorY2rkG9GWTa\n"
+                    + "qx40Tz9df9w8IEUrZLZqgdzLOTf/O0bzUwkn3UwvSGUrC1CUeEAUcYqeIXd9IzPe\n"
+                    + "5NLxgAC02MTtgddqTS1UKb1pAoGANgYAw/utQywzTRie4CfFQZXj8OM78yte1wV4\n"
+                    + "3/Zc6C2y647NguqEkYchEF75MwEPAGCg1Na6F6i5mU/0aJ5ym8EF21ikxlPnB0rn\n"
+                    + "Cb+kHE7HHs9aoyRhBY9FcTgqDDZAq38kprVPYN+rzGrwVK2S2Dm/tuXP6FkabuD8\n"
+                    + "dr6qJJECgYBKc09htDe/OBPqyVtEuUv0HTA5KRNpqeheWCKTDAFtjpciYa1qLyjs\n"
+                    + "JET3gPfjWkcuDOCs6zcaj/anlr/PxdSxYw4Zu21YKZGY70ceBEJgtX5PcrgmpH7R\n"
+                    + "99+DOAhlNqwK5X+4Foo3tErRveaTczgz38G17IfoKJ7rKYHQt1o+bQ==\n" + "-----END RSA PRIVATE KEY-----";
+            bw.write(validKey);
+        } catch (Exception e) {
+            throw e;
+        }
+        System.setProperty(MANTA_KEY_PATH_PROPERTY, keyFile.getAbsolutePath());
+        MantaClient client = MANTA_CLIENT_SUPPLIER.get();
+        Stream<MantaObject> mor = client.listObjects("/" + System.getProperty(MANTA_USER) + "/stor");
+        Iterator<MantaObject> it = mor.iterator();
+        while (it.hasNext()) {
+            Log.info(it.next());
+        }
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = ".*protocol = null.*", description = "This will use an invalid URL")
+    public void invalidURL() throws IOException {
+        System.setProperty(MANTA_URL_PROPERTY, "www.google.com");
+        MantaClient client = MANTA_CLIENT_SUPPLIER.get();
+        Stream<MantaObject> mor = client.listObjects(base_dir);
+        Assert.fail(String.format("Successfully made a call that we should not be able to call, found : %d objects",
+                mor.count()));
+    }
+
+    @Test
+    public void emptyURL() {
+        System.clearProperty(MANTA_URL_PROPERTY);
+        MantaClient client = MANTA_CLIENT_SUPPLIER.get();
+        // Should default to us-east.
+        Assert.assertEquals(client.getContext().getMantaURL(), "https://us-east.manta.joyent.com");
+    }
+
+    @Test(expectedExceptions = MantaClientHttpResponseException.class, expectedExceptionsMessageRegExp = ".*HTTP request failed to.*")
+    public void invalidUser() throws IOException {
+        System.setProperty(MANTA_USER_PROPERTY, "invalid");
+        MantaClient client = MANTA_CLIENT_SUPPLIER.get();
+        Stream<MantaObject> mor = client.listObjects(base_dir);
+        Assert.assertTrue(mor.count() > 0, "Seeing more than 0 entries, This should have thrown an exception by now.");
+    }
+
+    @Test(expectedExceptions = MantaClientHttpResponseException.class, expectedExceptionsMessageRegExp = ".*HTTP request failed to:.*")
+    public void invalidPassword() throws IOException {
+        System.setProperty(MANTA_USER_PROPERTY, "invalid");
```

Comment:
This test is about password but this line seems to be setting the username? Changing it to "manta.password" triggers this exception which is probably what you expected to be testing:

```
com.joyent.http.signature.KeyLoadException: Loaded key is not encrypted but a password was supplied.
```

Either way, password are used when loading keys, I have no idea why you would expect a `MantaClientHttpResponseException` for a bad password. You'll also need an encrypted private key to properly test the password setting.